### PR TITLE
coverity fixes2

### PR DESCRIPTION
**Pull request by rgetz**
fix the remaining outstanding/recently added coverity issues, and one that has been there for awhile.

### CODE REVIEW

File: tests/iio_stresstest.c [hunk 1]
```diff
@@ -385,15 +385,18 @@ int main(int argc, char **argv)
 			break;
 		case 'b':
 			info.arg_index += 2;
-			info.buffer_size = atoi(info.argv[info.arg_index]);
+			info.buffer_size = strtol(info.argv[info.arg_index], NULL, 10);
+			/* Max 4M */
+			if (info.buffer_size > (1024 * 1024 * 4))
+				info.buffer_size = 1024 * 1024 * 4;
```
Comment (commodo) [inline tests/iio_stresstest.c:391]: this max looks like slightly new behavior;
but it's fine to leave it here

File: iiod/iiod.c [hunk 2]
```diff
@@ -407,9 +407,24 @@ static int main_interactive(struct iio_context *ctx, bool verbose, bool use_aio)
 
 	if (!use_aio) {
 		flags = fcntl(STDIN_FILENO, F_GETFL);
-		fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags > 0)
```
Comment (commodo) [inline iiod/iiod.c:410]: is `flags >= 0`  also valid? i.e. `flags` being also zero;

File: iiod/iiod.c [hunk 3]
```diff
@@ -407,9 +407,24 @@ static int main_interactive(struct iio_context *ctx, bool verbose, bool use_aio)
...
+			char err_str[1024];
+			iio_strerror(errno, err_str, sizeof(err_str));
+			ERROR("Could not get/set O_NONBLOCK on STDIN_FILENO"
+					" %s (%d)\n", err_str, -errno);
+		}
+
 		flags = fcntl(STDOUT_FILENO, F_GETFL);
-		fcntl(STDOUT_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags > 0)
+			flags = fcntl(STDOUT_FILENO, F_SETFL, flags | O_NONBLOCK);
```
Comment (commodo) [inline iiod/iiod.c:421]: same question here about `flags >= 0`
thing is: depending on the case, `flags >= 0` can change behavior or introduce a new bug

File: iiod/iiod.c [hunk 4]
```diff
@@ -407,9 +407,24 @@ static int main_interactive(struct iio_context *ctx, bool verbose, bool use_aio)
 
 	if (!use_aio) {
 		flags = fcntl(STDIN_FILENO, F_GETFL);
-		fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags > 0)
+			flags = fcntl(STDIN_FILENO, F_SETFL, flags | O_NONBLOCK);
+		if (flags < 0) {
+			char err_str[1024];
+			iio_strerror(errno, err_str, sizeof(err_str));
+			ERROR("Could not get/set O_NONBLOCK on STDIN_FILENO"
```
Comment (dNechita) [inline iiod/iiod.c:415]: ERROR needs to be updated to: IIO_ERROR. Maybe rebase this onto master.

File: tests/iio_stresstest.c (hunk 1 above)
Comment (rgetz) [reply tests/iio_stresstest.c:391]: yes, it was complaining that the value was tainted (not checked), so check it....

File: iiod/iiod.c (hunk 2 above)
Comment (rgetz) [reply iiod/iiod.c]: I'm not sure if that is practical for STDIN (it should be open as write), but in theory, yes - it's not an error, so I updated. Good catch. Thanks

File: iiod/iiod.c (hunk 3 above)
Comment (rgetz) [reply iiod/iiod.c:421]: fixed / Thanks

File: iiod/iiod.c (hunk 4 above)
Comment (rgetz) [reply iiod/iiod.c]: yeah, rebased on master.

**Comment by rgetz** [comment]
Fixed 2.

image

**Pull request merged**